src/components/learner-credit-management/data/hooks/useBudgetDetailPage.js

```diff
@@ -0,0 +1,76 @@
+import { getBudgetStatus } from '../utils';
+import { BUDGET_STATUSES } from '../../../EnterpriseApp/data/constants';
+
+const transformOfferToPolicy = (offer) => {
+  if (!offer) { return null; }
+
+  return {
+    displayName: null,
+    subsidyActiveDatetime: offer.dateCreated,
+    subsidyExpirationDatetime: offer.exportTimestamp,
+    aggregates: {
+      spendAvailableUsd: offer.remainingBalance,
+      amountAllocatedUsd: 0,
+      amountRedeemedUsd: offer.amountOfOfferSpent,
+    },
+    spendLimit: offer.maxDiscount * 100,
+    isAssignable: false,
+  };
+};
+
+const assignBudgetStatus = (policy) => {
+  const {
+    status, badgeVariant, term, date,
+  } = getBudgetStatus(
+    policy.subsidyActiveDatetime,
+    policy.subsidyExpirationDatetime,
+  );
+
+  return {
+    status, badgeVariant, term, date,
+  };
+};
+
+const assignBudgetDetails = (policy) => {
+  const { spendAvailableUsd, amountAllocatedUsd, amountRedeemedUsd } = policy.aggregates;
+
+  const available = spendAvailableUsd;
+  const limit = policy.spendLimit / 100;
+  const utilized = policy.isAssignable
+    ? (amountAllocatedUsd + amountRedeemedUsd)
+    : amountRedeemedUsd;
+
+  return { budgetTotalSummary: { available, limit, utilized } };
+};
+
+const useBudgetDetailPage = (subsidyAccessPolicy, enterpriseOffer, budgetId) => {
+  const policy = subsidyAccessPolicy || transformOfferToPolicy(enterpriseOffer);
+
+  const defaultData = {
+    budgetId,
+    budgetTotalSummary: {
+      available: 0,
+      utilized: 0,
+      limit: 0,
+    },
+    budgetAggregates: policy?.aggregates || {},
+    badgeVariant: 'warning',
+    status: BUDGET_STATUSES.scheduled,
+    term: 'Ends',
+    date: '',
+    isAssignable: policy.isAssignable || false,
```

Comment:
Similar to the below comment about `!policy`, if `policy` can be undefined/null, then this line may throw a JS error in that case since it assumes `policy` always exists, unlike line 62 below which uses the `?.` optional chaining syntax.